test/client.js

```diff
@@ -438,6 +439,34 @@ describe('CloudantClient', function() {
           done();
         });
       });
+
+      it('performs request with piped payload and returns response', function(done) {
+        var mocks = nock(SERVER)
+            .post(DBNAME + '/_all_docs')
+            .reply(200, { rows: [{ key: 'doc1', error: 'not_found' }] });
+
+        var cloudantClient = new Client({ plugins: [] });
+        assert.equal(cloudantClient._plugins.length, 0);
+
+        var readable = new stream.Readable();
+        readable.push('{"keys":["doc1"]}'); // request payload
+        readable.push(null);
+
+        var options = {
+          url: SERVER + DBNAME + '/_all_docs',
+          auth: { username: ME, password: PASSWORD },
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' }
+        };
+
+        readable.pipe(cloudantClient.request(options, function(err, resp, data) {
+          assert.equal(err, null);
+          assert.equal(resp.statusCode, 200);
+          assert.ok(data.indexOf('"key":"doc1","error":"not_found"') > -1);
```

Comment:
Should there be an assert that `'{"keys":["doc1"]}'` was written to the mock server?

Reply:
yeah I guess. I couldn't be bothered to add the @before func that creates the doc so was testing for the not_found. but since you've mentioned it...

Reply:
ah, I've pinged  André Rouél it seems. apols. he must get that a lot!

Reply:
Fixed in bf83319.